app/presenters/nav_tab/link_collection.rb

```diff
@@ -0,0 +1,123 @@
+class NavTab::LinkCollection
+  include Rails.application.routes.url_helpers
+  include TranslationHelper
+
+  attr_reader :ability, :facility
+
+  def initialize(facility, ability)
+    @facility = facility
+    @ability = ability
+  end
+
+  def admin
+    default + [
+      admin_orders,
+      admin_reservations,
+      admin_billing,
+      admin_products,
+      admin_users,
+      admin_reports,
+      admin_facility,
+    ].select(&:present?)
+  end
+
+  def customer
+    default + [orders, reservations, accounts]
+  end
+
+  def default
+    [home]
+  end
+
+  private
+
+  def accounts
+    NavTab::Link.new(tab: :accounts, text: t_my(Account), url: accounts_path)
+  end
+
+  def admin_billing
+    if !facility.all_facility? && ability.can?(:manage_billing, facility)
```

Comment:
Would it be worth extracting

```ruby
def single_facility?
  !facility.all_facility?
end
```

It's a bunch of places in this class and would probably be easier to read.